komga-webui/src/components/dialogs/EditSeriesDialog.vue

```diff
@@ -611,6 +656,31 @@ export default Vue.extend({
       return null
     },
     async editSeries(): Promise<boolean> {
+      if (this.single && this.uploadQueue.length > 0) {
+        const name = this.selectedThumbnail
+        const series = this.series as SeriesDto
+        for (const file of this.uploadQueue) {
+          const body = new FormData()
+          body.append('file', file)
+          body.append('selected', `${file.name === name}`)
+          await this.$komgaSeries.uploadThumbnail(series.id, body)
+        }
+      }
+
+      if (this.single && this.selectedThumbnail !== '') {
+        const id = this.selectedThumbnail
+        const series = this.series as SeriesDto
+        if (this.seriesThumbnails.find(value => value.id === id)) {
+          await this.$komgaSeries.markThumbnailAsSelected(series.id, id)
+        }
+      }
+
+      if (this.single && this.deleteQueue.length > 0) {
+        for (const toDelete of this.deleteQueue) {
+          await this.$komgaSeries.deleteThumbnail(toDelete.seriesId, toDelete.id)
```

Comment:
same here, do we need to await here ? that would block the other requests while waiting